### PR TITLE
Adding default expanded option to commandline and config (closes #822)

### LINKED ISSUE #822: Ability to launch a specific widget already maximized, without having to press "e" manually
### Pretty much as the title says

Ability to launch a specific widget already maximized, without having to press "e" manually after selecting one of the widgets.

I see that -e is not an existing command line option, so you could make it so that the following command would load Bottom's CPU widget:

btm -e cpu

As you can see in the screenshot, I like to run 3 instances of Bottom in one Konsole window that is split vertically into 3 transparent terminals (I have configured it so that I can hide the title bar with a keyboard shortcut). I'd like to automate loading these as much as I can, so being able to load Bottom widgets directly from the command line would be a great help.

Screenshot_20221002_113359

And apart form my own silly UI customizations, I'm sure other users would benefit from it.

### PR DESCRIPTION
**Pull request by sou-chon**
## Description

_A description of the change, what it does, and why it was made. If relevant (such as any change that modifies the UI), **please provide screenshots** of the changes:_

Addresses #822: Adding option to expand widgets on startup. Also adding the corresponding option to config file.

## Issue

_If applicable, what issue does this address?_

Closes: #822 

## Testing

_If relevant, please state how this was tested. All changes **must** be tested to work:_

testing by `debug_assert()` from `clap`.

_If this is a code change, please also indicate which platforms were tested:_

- [ ] _Windows_
- [x] _macOS_
- [x] _Linux_

## Checklist

_If relevant, ensure the following have been met:_

- [x] _Areas your change affects have been linted using rustfmt (`cargo fmt`)_
- [x] _The change has been tested and doesn't appear to cause any unintended breakage_
- [x] _Documentation has been added/updated if needed (`README.md`, help menu, doc pages, etc.)_
- [x] _The pull request passes the provided CI pipeline_
- [x] _There are no merge conflicts_
- [ ] _If relevant, new tests were added (don't worry too much about coverage)_

### CODE REVIEW

**Comment by ClementTsang** [comment]
Thanks for the PR, will check in a bit.

File: src/clap.rs [hunk 1]
```diff
@@ -316,6 +316,11 @@ use CPU (3) as the default instead.
         .help("Sets the default widget type, use --help for info.")
         .long_help(DEFAULT_WIDGET_TYPE_STR);
 
+    let expanded_on_startup = Arg::new("expanded_on_startup")
+        .long("expanded")
+        .short('e')
+        .help("Expanded selected widget upon starting the app. Same as pressing \"e\" inside the app. Use with \"default_widget_type\" and \"default_widget_count\" to select desired expanded widget.");
```
Comment (ClementTsang) [inline src/clap.rs:322]: nit: would move this to `long_help`, I usually prefer to try to keep `help` shorter and concise.

File: src/constants.rs [hunk 2]
```diff
@@ -518,6 +518,7 @@ pub const CONFIG_TEXT: &str = r##"# This is a default config file for bottom.  A
 # Override layout default widget
 #default_widget_type = "proc"
 #default_widget_count = 1
+#expanded_on_startup = false
```
Comment (ClementTsang) [inline src/constants.rs:521]: nit: add a comment on what this option does; feel free to just reuse your clap description. Also would be nice if you update `sample_configs/default_config.toml`.

File: src/options.rs [hunk 3]
```diff
@@ -172,7 +173,7 @@ pub struct IgnoreList {
 pub fn build_app(
     matches: &ArgMatches, config: &mut Config, widget_layout: &BottomLayout,
     default_widget_id: u64, default_widget_type_option: &Option<BottomWidgetType>,
-    colours: &CanvasColours,
+    expanded_upon_startup: bool, colours: &CanvasColours,
```
Comment (ClementTsang) [inline src/options.rs:176]: nit: Could also just grab `expanded_upon_startup` within this function instead, I guess, as opposed to passing it around.

**Review by ClementTsang — CHANGES_REQUESTED** [review]
Sorry for taking so long, left a few small comments, mostly just nitpicking.

EDIT: Also did a quick check on my Linux machine for sanity's sake, seems fine functionality-wise.

**Comment by alexleduc76** [comment]
Thank you for this! 

Will you make a new release soon? Just wondering if I should clone the branch and build it myself, or wait for an official release (if one of coming soon).

**Comment by ClementTsang** [comment]
@alexleduc76 I want to get an official stable release out before end of year.

That being said I have daily nightly builds automatically created that might be of interest at https://github.com/ClementTsang/bottom/releases/nighty

**Comment by sou-chon** [comment]
@ClementTsang updated

File: src/options.rs [hunk 4]
```diff
@@ -419,6 +422,7 @@ pub fn build_app(
         .current_widget(widget_map.get(&initial_widget_id).unwrap().clone()) // TODO: [UNWRAP] - many of the unwraps are fine (like this one) but do a once-over and/or switch to expect?
         .widget_map(widget_map)
         .used_widgets(used_widgets)
+        .is_expanded(expanded_upon_startup)
```
Comment (ClementTsang) [inline src/options.rs:425]: Sorry for not catching this earlier, but we probably want to make this not fire if in basic mode (e.g. `btm --basic -e`), as expansion doesn't do anything in that mode at the moment.

**Review by ClementTsang — CHANGES_REQUESTED** [review]

File: src/options.rs (hunk 4 above)
Comment (sou-chon) [reply src/options.rs]: @ClementTsang  good shout. updated now

**Review by ClementTsang — APPROVED** [review]

**Pull request merged**